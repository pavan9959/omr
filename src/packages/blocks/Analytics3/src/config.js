Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.validationApiContentType = "application/json";
exports.validationApiMethodType = "GET";
exports.exampleAPiEndPoint = "EXAMPLE_URL";
exports.exampleAPiMethod = "POST";
exports.exampleApiContentType = "application/json";
exports.textInputPlaceHolder = "Enter Text";
exports.labelTitleText = "Analytics3";
exports.labelBodyText = "Analytics3 Body";

exports.btnExampleTitle = "CLICK ME";
// Customizable Area End