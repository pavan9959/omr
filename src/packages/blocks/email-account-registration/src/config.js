Object.defineProperty(exports, "__esModule", {
  value: true
});

exports.placeHolderEmail = "Email";

exports.labelHeader = "Value proposition: why users should sign up.";
exports.labelFirstName = "First name";
exports.lastName = "Last name";
exports.labelEmail = "Email";
exports.labelPassword = "Password";
exports.labelRePassword = "Re-Type Password";
exports.errorCountryCodeNotSelected = "Please select country code";
exports.errorBlankField = "can't be blank";

exports.labelLegalText =
  "Legal text: By signing up you agree to our Terms &amp; conditions and Privacy policy.";

exports.labelLegalTermCondition = "Terms and conditions";
exports.labelLegalPrivacyPolicy = "Privacy policy";
exports.btnTextSignUp = "Sign up";
exports.errorPasswordNotValid = "Password not valid.";

exports.errorTitle = "Error";

exports.errorBothPasswordsNotSame = "Passwords should be same.";
exports.errorAllFieldsAreMandatory = "All fields are mandatory.";

exports.contentTypeApiAddDetail = "application/json";

exports.accountsAPiEndPoint =
  "account/accounts";

exports.apiMethodTypeAddDetail = "POST";

exports.errorEmailNotValid = "Email not valid.";
exports.urlGetValidations =
  "profile/validations";
exports.validationApiContentType = "application/json";

exports.validationApiMethodType = "GET";
