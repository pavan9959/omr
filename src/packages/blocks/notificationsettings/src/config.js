Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.httpGetMethod = "GET";
exports.httpPatchMethod =  "PATCH";

exports.getNotificationAPIEndPoint = "notifsettings/notification_settings";
exports.updateNotificationAPIEndPoint = "notifsettings/notification_settings/";
exports.updateNotificationGroupAPIEndPoint = "notifsettings/groups/";
exports.updateNotificationSubGroupAPIEndPoint = "notifsettings/subgroups/";

exports.notificationApiContentType = "application/json";

// Customizable Area End