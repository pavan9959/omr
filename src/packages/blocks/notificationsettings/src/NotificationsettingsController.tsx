import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  txtInputValue: string;
  txtSavedValue: string;
  enableField: boolean;
  checkedActionBar: boolean;
  checkedActionBarText: string;
  notificationData: any;
  token: string;

  // Customizable Area End
}

interface SS {
  id: any;
}

export default class NotificationsettingsController extends BlockComponent<
  Props,
  S,
  SS
> {
  notificationAPICallId: any;
  setNotificationAPICall: any;
  setNotificationGroupAPICall: any;
  setNotificationSubGroupAPICall: any;

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [
      getName(MessageEnum.SessionResponseMessage),
      getName(MessageEnum.NavigationPayLoadMessage),
      getName(MessageEnum.RestAPIResponceMessage)
    ];

    this.state = {
      txtInputValue: "",
      txtSavedValue: "A",
      enableField: false,
      checkedActionBar: false,
      checkedActionBarText: configJSON.notificationGroup1,
      notificationData: [],
      token: ""
    };
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
  async componentDidMount() {
    super.componentDidMount();
    this.getToken();
    if (this.isPlatformWeb() === false) {
      this.props.navigation.addListener("willFocus", () => {
        this.getToken();
      });
    }
  }

  getToken = () => {
    const msg: Message = new Message(
      getName(MessageEnum.SessionRequestMessage)
    );
    this.send(msg);
  };

  async receive(from: string, message: Message) {
    // Customizable Area Start
    if (getName(MessageEnum.SessionResponseMessage) === message.id) {
      runEngine.debugLog("Message Recived", message);

      let token = message.getData(getName(MessageEnum.SessionResponseToken));
      this.setState({ token: token });
      this.getNotificationDataRequest(token);
    } else if (getName(MessageEnum.RestAPIResponceMessage) === message.id) {
      const apiRequestCallId = message.getData(
        getName(MessageEnum.RestAPIResponceDataMessage)
      );

      var responseJson = message.getData(
        getName(MessageEnum.RestAPIResponceSuccessMessage)
      );

      var errorReponse = message.getData(
        getName(MessageEnum.RestAPIResponceErrorMessage)
      );
      if (responseJson.data) {
        if (apiRequestCallId === this.notificationAPICallId) {
          this.setState({ notificationData: responseJson.data });
        }
        if (apiRequestCallId === this.setNotificationAPICall) {
          var array = this.state.notificationData;
          for (let i = 0; i < array.length; i++) {
            if (array[i].id === responseJson.data.id) {
              array[i] = responseJson.data;
            }
          }
          this.setState({ notificationData: array });
        }
        if (apiRequestCallId === this.setNotificationGroupAPICall) {
          var array = this.state.notificationData;
          for (let i = 0; i < array.length; i++) {
            for (
              let j = 0;
              j < array[i].attributes.notification_groups.data.length;
              j++
            ) {
              if (
                array[i].attributes.notification_groups.data[j].id ===
                responseJson.data.id
              ) {
                array[i].attributes.notification_groups.data[j] =
                  responseJson.data;
              }
            }
          }
          this.setState({ notificationData: array });
        }
        if (apiRequestCallId === this.setNotificationSubGroupAPICall) {
          var array = this.state.notificationData;
          for (let i = 0; i < array.length; i++) {
            for (
              let j = 0;
              j < array[i].attributes.notification_groups.data.length;
              j++
            ) {
              for (
                let k = 0;
                k <
                array[i].attributes.notification_groups.data[j].attributes
                  .notification_subgroups.data.length;
                k++
              ) {
                if (
                  array[i].attributes.notification_groups.data[j].attributes
                    .notification_subgroups.data[k].id === responseJson.data.id
                ) {
                  array[i].attributes.notification_groups.data[
                    j
                  ].attributes.notification_subgroups.data[k] =
                    responseJson.data;
                }
              }
            }
          }
          this.setState({ notificationData: array });
        }
        runEngine.debugLog("API Recived", responseJson.data);
      } else if (responseJson.errors) {
        this.parseApiErrorResponse(responseJson);
        this.setState({ notificationData: this.state.notificationData });
        this.parseApiCatchErrorResponse(errorReponse);
      }
    }
    // Customizable Area End
  }

  getNotificationDataRequest = (token: string) => {
    const header = {
      "Content-Type": configJSON.notificationApiContentType,
      token: token
    };
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    this.notificationAPICallId = requestMessage.messageId;

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.getNotificationAPIEndPoint
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.httpGetMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);
  };

  // Customizable Area Start

  setMainCheckBox(item: any) {
    var data = {
      title: item.attributes.title,
      description: item.attributes.description,
      state: item.attributes.state === "active" ? "inactive" : "active"
    };
    this.postNotificationDataRequest(
      configJSON.updateNotificationAPIEndPoint + `${item.id}`,
      data,
      "setNotification"
    );
  }

  setGroupCheckBox(index: number, groupIndex: number, groupitem: any) {
    var data = {
      group_type: groupitem.attributes.group_type,
      group_name: groupitem.attributes.group_name,
      state: groupitem.attributes.state === "active" ? "inactive" : "active",
      notification_setting_id: groupitem.attributes.notification_setting_id
    };
    this.postNotificationDataRequest(
      configJSON.updateNotificationGroupAPIEndPoint + `${groupitem.id}`,
      data,
      "setGroupNotification"
    );
  }

  setSubGroupCheckBox = (subGroupItem: any) => {
    let data = {
      subgroup_type: subGroupItem.attributes.subgroup_type,
      subgroup_name: subGroupItem.attributes.subgroup_name,
      state: subGroupItem.attributes.state === "active" ? "inactive" : "active",
      notification_group_id: subGroupItem.attributes.notification_group_id
    };

    this.postNotificationDataRequest(
      configJSON.updateNotificationSubGroupAPIEndPoint + `${subGroupItem.id}`,
      data,
      "setSubGroupNotification"
    );
  };

  postNotificationDataRequest = (
    endPointURL: string,
    data: any,
    type: string
  ): boolean => {
    const header = {
      "Content-Type": configJSON.notificationApiContentType,
      token: this.state.token
    };

    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );

    if (type === "setNotification") {
      this.setNotificationAPICall = requestMessage.messageId;
    } else if (type === "setGroupNotification") {
      this.setNotificationGroupAPICall = requestMessage.messageId;
    } else if (type === "setSubGroupNotification") {
      this.setNotificationSubGroupAPICall = requestMessage.messageId;
    }

    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      endPointURL
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestBodyMessage),
      JSON.stringify(data)
    );

    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.httpPatchMethod
    );

    runEngine.sendMessage(requestMessage.id, requestMessage);
    return true;
  };

  // Customizable Area End
}
