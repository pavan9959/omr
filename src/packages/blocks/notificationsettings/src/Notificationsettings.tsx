import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableWithoutFeedback,
  Platform
} from "react-native";
//@ts-ignore
import CustomCheckBox from "../../../components/src/CustomCheckBox";
// Customizable Area End

import NotificationsettingsController, {
  Props
} from "./NotificationsettingsController";

export default class Notificationsettings extends NotificationsettingsController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  // Customizable Area End

  render() {
    return (
      //Merge Engine DefaultContainer
      <ScrollView keyboardShouldPersistTaps="always" style={styles.container}>
        <TouchableWithoutFeedback
          onPress={() => {
            this.hideKeyboard();
          }}
        >
          {/* Customizable Area Start */}
          {/* Merge Engine UI Engine Code */}

          <View>
            {this.state.notificationData.map((item: any, index: number) => {
              return (
                <View key={index}>
                  <View style={styles.innerView}>
                    <Text
                      testID={"notification" + index} //Merge Engine::From BDS
                      style={styles.title} //UI Engine::From Sketch
                    >
                      {item.attributes.title}
                    </Text>
                    <CustomCheckBox
                      testID={"CustomCheckBox" + index} //Merge Engine::From BDS
                      isChecked={
                        item.attributes.state === "active" ? true : false
                      }
                      onChangeValue={() => {
                        this.setMainCheckBox(item);
                      }}
                    />
                  </View>
                  {item.attributes.state === "active" &&
                    item.attributes.notification_groups.data.map(
                      (groupitem: any, groupindex: number) => {
                        return (
                          <View key={groupindex}>
                            <View style={styles.innerView}>
                              <Text
                                testID={"notificationGroup1" + groupindex} //Merge Engine::From BDS
                                style={styles.title} //UI Engine::From Sketch
                              >
                                {groupitem.attributes.group_name}
                              </Text>
                              <CustomCheckBox
                                testID={"CustomCheckBox" + groupindex} //Merge Engine::From BDS
                                isChecked={
                                  groupitem.attributes.state === "active"
                                    ? true
                                    : false
                                }
                                onChangeValue={() => {
                                  this.setGroupCheckBox(
                                    index,
                                    groupindex,
                                    groupitem
                                  );
                                }}
                              />
                            </View>
                            {groupitem.attributes.state === "active" &&
                              groupitem.attributes.notification_subgroups.data.map(
                                (subgroupitem: any, subgroupindex: number) => {
                                  return (
                                    <View
                                      key={subgroupindex}
                                      style={styles.subInnerView}
                                    >
                                      <Text
                                        testID={
                                          "notificationSubGroup" + subgroupindex
                                        } //Merge Engine::From BDS
                                        style={styles.subTitle} //UI Engine::From Sketch
                                      >
                                        {subgroupitem.attributes.subgroup_name}
                                      </Text>
                                      <CustomCheckBox
                                        testID={
                                          "CustomCheckBox" + subgroupindex
                                        }
                                        isChecked={
                                          subgroupitem.attributes.state ===
                                          "active"
                                            ? true
                                            : false
                                        }
                                        onChangeValue={() => {
                                          this.setSubGroupCheckBox(
                                            subgroupitem
                                          );
                                        }}
                                      />
                                    </View>
                                  );
                                }
                              )}
                          </View>
                        );
                      }
                    )}
                </View>
              );
            })}
          </View>
          {/* Merge Engine UI Engine Code */}
          {/* Customizable Area End */}
        </TouchableWithoutFeedback>
      </ScrollView>
      //Merge Engine End DefaultContainer
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    backgroundColor: "#f7f9fa"
  },
  innerView: {
    flex: 1,
    padding: 16,
    marginLeft: "auto",
    marginRight: "auto",
    marginTop: 5,
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    flexDirection: "row",
    justifyContent: "space-between",
    backgroundColor: "#ffffffff"
  },
  subInnerView: {
    flex: 1,
    paddingHorizontal: 16,
    paddingVertical: 10,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    flexDirection: "row",
    justifyContent: "space-between",
    backgroundColor: "#ffffffff"
  },
  title: {
    // marginBottom: 32,
    fontSize: 16,
    textAlign: "left",
    marginVertical: 4
  },
  subTitle: {
    // marginBottom: 32,
    fontSize: 14,
    textAlign: "left",
    marginVertical: 4
  },
  body: {
    marginBottom: 32,
    fontSize: 16,
    textAlign: "left",
    marginVertical: 8
  },
  bgPasswordContainer: {
    flexDirection: "row",
    backgroundColor: "#00000000",
    marginBottom: 16,
    borderBottomWidth: 1,
    borderColor: "#767676",
    borderRadius: 2,
    padding: 10,
    borderWidth: Platform.OS === "web" ? 0 : 1
  },
  bgMobileInput: {
    flex: 1
  },
  showHide: {
    alignSelf: "center"
  },
  imgShowhide: Platform.OS === "web" ? { height: 30, width: 30 } : {}
});
// Customizable Area End
