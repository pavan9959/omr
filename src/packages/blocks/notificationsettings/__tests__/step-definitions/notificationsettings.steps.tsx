import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'

import * as helpers from '../../../../framework/src/Helpers'
import {runEngine} from '../../../../framework/src/RunEngine'
import {Message} from "../../../../framework/src/Message"
export const configJSON = require("../../config.json");

import MessageEnum, {getName} from "../../../../framework/src/Messages/MessageEnum"; 
import React from "react";
import Notificationsettings from "../../src/Notificationsettings"
const navigation = require("react-navigation")

const screenProps = {
    navigation: navigation,
    id: "Notificationsettings"
  }

const feature = loadFeature('./__tests__/features/notificationsettings-scenario.feature');

defineFeature(feature, (test) => {


    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
    });

    test('User navigates to notificationsettings', ({ given, when, then }) => {
        let notificationSettingsWrapper:ShallowWrapper;
        let instance:Notificationsettings; 

        given('I am a User loading notificationsettings', () => {
            notificationSettingsWrapper = shallow(<Notificationsettings {...screenProps}/>)
            expect(notificationSettingsWrapper).toBeTruthy()
            expect(notificationSettingsWrapper).toMatchSnapshot()     

            instance = notificationSettingsWrapper.instance()as Notificationsettings;

            const getNotificationAPI = new Message(getName(MessageEnum.RestAPIResponceMessage))
            getNotificationAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), getNotificationAPI.messageId);
            getNotificationAPI.addData(getName(MessageEnum.RestAPIResponceSuccessMessage), 
            {
                "data": [
                    {
                        "id": "2",
                        "type": "notification_settings",
                        "attributes": {
                            "title": "Settings #1 update",
                            "description": "Description updated",
                            "state": "active",
                            "notification_groups": {
                                "data": [
                                    {
                                        "id": "3",
                                        "type": "notification_groups",
                                        "attributes": {
                                            "group_type": "order_group",
                                            "group_name": "group #3",
                                            "notification_setting_id": 2,
                                            "state": "active",
                                            "notification_subgroups": {
                                                "data": [
                                                    {
                                                        "id": "2",
                                                        "type": "notification_subgroups",
                                                        "attributes": {
                                                            "subgroup_type": "order_shipped",
                                                            "subgroup_name": "name 4",
                                                            "notification_group_id": 3,
                                                            "state": "inactive"
                                                        }
                                                    }
                                                ]
                                            }
                                        }
                                    }
                                ]
                            }
                        }
                    }
                ]
            });
            instance.notificationAPICallId = getNotificationAPI.messageId
            runEngine.sendMessage("Unit Test", getNotificationAPI)

        });

        when('I navigate to the notificationsettings', () => {
             instance = notificationSettingsWrapper.instance() as Notificationsettings
        });

        then('notificationsettings will load with out errors', () => {
            expect(notificationSettingsWrapper).toBeTruthy()
            expect(notificationSettingsWrapper).toMatchSnapshot()
            
        });

        then('I can leave the screen with out errors', () => {
            instance.componentWillUnmount()
            expect(notificationSettingsWrapper).toBeTruthy()
            expect(notificationSettingsWrapper).toMatchSnapshot()
        });
    });

    test('Set notification settings active or inactive', ({ given, when, then }) => {
        let notificationSettingsWrapper:ShallowWrapper;
        let instance:Notificationsettings; 

        given('I am a User attempting to set notification settings active or inactive', () => {
            notificationSettingsWrapper = shallow(<Notificationsettings {...screenProps}/>)
            expect(notificationSettingsWrapper).toBeTruthy()
            expect(notificationSettingsWrapper).toMatchSnapshot()
        });

        when('I set notification settings active or inactive', () => {
             instance = notificationSettingsWrapper.instance() as Notificationsettings
            //  instance.setState({email: "", password: "password!"})
        });

        then('notification settings should set active or inactive', () => {
            var data={
                "title": "Settings #1 update",
                "description": "Description updated",
                "state": "inactive"
            }
            expect(instance.postNotificationDataRequest(configJSON.updateNotificationAPIEndPoint+`2`,data,"setNotification")).toBe(true);

            const msgSetNotificationAPI = new Message(getName(MessageEnum.RestAPIResponceMessage))
            msgSetNotificationAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), msgSetNotificationAPI);
            msgSetNotificationAPI.addData(getName(MessageEnum.RestAPIResponceSuccessMessage), 
            {
                "data": {
                    "id": "2",
                    "type": "notification_settings",
                    "attributes": {
                        "title": "Settings #1 update",
                        "description": "Description updated",
                        "state": "inactive",
                        "notification_groups": {
                            "data": [
                                {
                                    "id": "3",
                                    "type": "notification_groups",
                                    "attributes": {
                                        "group_type": "order_group",
                                        "group_name": "group #3",
                                        "notification_setting_id": 2,
                                        "state": "inactive",
                                        "notification_subgroups": {
                                            "data": [
                                                {
                                                    "id": "2",
                                                    "type": "notification_subgroups",
                                                    "attributes": {
                                                        "subgroup_type": "order_shipped",
                                                        "subgroup_name": "name 4",
                                                        "notification_group_id": 3,
                                                        "state": "inactive"
                                                    }
                                                }
                                            ]
                                        }
                                    }
                                }
                            ]
                        }
                    }
                }
            });

            msgSetNotificationAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), msgSetNotificationAPI.messageId);
            instance.setNotificationAPICall = msgSetNotificationAPI.messageId
            runEngine.sendMessage("Unit Test", msgSetNotificationAPI);
        });
    });

    test('Set notification settings group active or inactive', ({ given, when, then }) => {
        let notificationSettingsWrapper:ShallowWrapper;
        let instance:Notificationsettings; 

        given('I am a User attempting to set notification settings group active or inactive', () => {
            notificationSettingsWrapper = shallow(<Notificationsettings {...screenProps}/>)
            expect(notificationSettingsWrapper).toBeTruthy()
            expect(notificationSettingsWrapper).toMatchSnapshot()
        });

        when('I set notification settings group active or inactive', () => {
             instance = notificationSettingsWrapper.instance() as Notificationsettings
            //  instance.setState({email: "", password: "password!"})
        });

        then('notification settings group should set active or inactive', () => {
            var data={
                "group_type": "order_group",
                "group_name": "group #3",
                "state": "active",
                "notification_setting_id": 2
            }
            expect(instance.postNotificationDataRequest(configJSON.updateNotificationGroupAPIEndPoint+`3`,data,"setGroupNotification")).toBe(true);

            const msgSetNotificationGroupAPI = new Message(getName(MessageEnum.RestAPIResponceMessage))
            msgSetNotificationGroupAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), msgSetNotificationGroupAPI);
            msgSetNotificationGroupAPI.addData(getName(MessageEnum.RestAPIResponceSuccessMessage), 
            {
                "data": {
                    "id": "3",
                    "type": "notification_groups",
                    "attributes": {
                        "group_type": "order_group",
                        "group_name": "group #3",
                        "notification_setting_id": 2,
                        "state": "active",
                        "notification_subgroups": {
                            "data": [
                                {
                                    "id": "2",
                                    "type": "notification_subgroups",
                                    "attributes": {
                                        "subgroup_type": "order_shipped",
                                        "subgroup_name": "name 4",
                                        "notification_group_id": 3,
                                        "state": "inactive"
                                    }
                                }
                            ]
                        }
                    }
                }
            });

            msgSetNotificationGroupAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), msgSetNotificationGroupAPI.messageId);
            instance.setNotificationGroupAPICall = msgSetNotificationGroupAPI.messageId
            runEngine.sendMessage("Unit Test", msgSetNotificationGroupAPI);
        });
    });

    test('Set notification settings sub group active or inactive', ({ given, when, then }) => {
        let notificationSettingsWrapper:ShallowWrapper;
        let instance:Notificationsettings; 

        given('I am a User attempting to set notification settings sub group active or inactive', () => {
            notificationSettingsWrapper = shallow(<Notificationsettings {...screenProps}/>)
            expect(notificationSettingsWrapper).toBeTruthy()
            expect(notificationSettingsWrapper).toMatchSnapshot()
        });

        when('I set notification settings sub group active or inactive', () => {
             instance = notificationSettingsWrapper.instance() as Notificationsettings
            //  instance.setState({email: "", password: "password!"})
        });

        then('notification settings sub group should set active or inactive', () => {
            var data={
                "subgroup_type": "order_shipped",
                "subgroup_name": "name 4",
                "notification_group_id": 3,
                "state": "active"
            }
            expect(instance.postNotificationDataRequest(configJSON.updateNotificationSubGroupAPIEndPoint+`2`,data,"setSubGroupNotification")).toBe(true);

            const msgSetNotificationSubGroupAPI = new Message(getName(MessageEnum.RestAPIResponceMessage))
            msgSetNotificationSubGroupAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), msgSetNotificationSubGroupAPI);
            msgSetNotificationSubGroupAPI.addData(getName(MessageEnum.RestAPIResponceSuccessMessage), 
            {
                "data": {
                    "id": "2",
                    "type": "notification_subgroups",
                    "attributes": {
                        "subgroup_type": "order_shipped",
                        "subgroup_name": "name 4",
                        "notification_group_id": 3,
                        "state": "active"
                    }
                }
            });

            msgSetNotificationSubGroupAPI.addData(getName(MessageEnum.RestAPIResponceDataMessage), msgSetNotificationSubGroupAPI.messageId);
            instance.setNotificationSubGroupAPICall = msgSetNotificationSubGroupAPI.messageId
            runEngine.sendMessage("Unit Test", msgSetNotificationSubGroupAPI);
        });
    });


});
