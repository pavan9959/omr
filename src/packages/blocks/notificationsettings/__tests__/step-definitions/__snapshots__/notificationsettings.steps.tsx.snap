// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notificationsettings Set notification settings active or inactive 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#f7f9fa",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View />
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`notificationsettings Set notification settings group active or inactive 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#f7f9fa",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View />
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`notificationsettings Set notification settings sub group active or inactive 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#f7f9fa",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View />
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`notificationsettings User navigates to notificationsettings 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#f7f9fa",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View />
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`notificationsettings User navigates to notificationsettings 2`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#f7f9fa",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <View
        key="0"
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffffff",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 5,
              "maxWidth": 650,
              "padding": 16,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
                "textAlign": "left",
              }
            }
            testID="notification0"
          >
            Settings #1 update
          </Text>
          <CustomCheckBox
            isChecked={true}
            onChangeValue={[Function]}
            testID="CustomCheckBox0"
          />
        </View>
        <View
          key="0"
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffffff",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": "auto",
                "marginRight": "auto",
                "marginTop": 5,
                "maxWidth": 650,
                "padding": 16,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginVertical": 4,
                  "textAlign": "left",
                }
              }
              testID="notificationGroup10"
            >
              group #3
            </Text>
            <CustomCheckBox
              isChecked={true}
              onChangeValue={[Function]}
              testID="CustomCheckBox0"
            />
          </View>
          <View
            key="0"
            style={
              Object {
                "backgroundColor": "#ffffffff",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": "auto",
                "marginRight": "auto",
                "maxWidth": 650,
                "paddingHorizontal": 16,
                "paddingVertical": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "marginVertical": 4,
                  "textAlign": "left",
                }
              }
              testID="notificationSubGroup0"
            >
              name 4
            </Text>
            <CustomCheckBox
              isChecked={false}
              onChangeValue={[Function]}
              testID="CustomCheckBox0"
            />
          </View>
        </View>
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`notificationsettings User navigates to notificationsettings 3`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#f7f9fa",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <View
        key="0"
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffffff",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 5,
              "maxWidth": 650,
              "padding": 16,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginVertical": 4,
                "textAlign": "left",
              }
            }
            testID="notification0"
          >
            Settings #1 update
          </Text>
          <CustomCheckBox
            isChecked={true}
            onChangeValue={[Function]}
            testID="CustomCheckBox0"
          />
        </View>
        <View
          key="0"
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffffff",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": "auto",
                "marginRight": "auto",
                "marginTop": 5,
                "maxWidth": 650,
                "padding": 16,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginVertical": 4,
                  "textAlign": "left",
                }
              }
              testID="notificationGroup10"
            >
              group #3
            </Text>
            <CustomCheckBox
              isChecked={true}
              onChangeValue={[Function]}
              testID="CustomCheckBox0"
            />
          </View>
          <View
            key="0"
            style={
              Object {
                "backgroundColor": "#ffffffff",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": "auto",
                "marginRight": "auto",
                "maxWidth": 650,
                "paddingHorizontal": 16,
                "paddingVertical": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "marginVertical": 4,
                  "textAlign": "left",
                }
              }
              testID="notificationSubGroup0"
            >
              name 4
            </Text>
            <CustomCheckBox
              isChecked={false}
              onChangeValue={[Function]}
              testID="CustomCheckBox0"
            />
          </View>
        </View>
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;
